src/Bicep.Core/PrettyPrint/DocumentBuildVisitor.cs

```diff
@@ -445,6 +445,42 @@ public override void VisitSyntaxTrivia(SyntaxTrivia syntaxTrivia)
                 this.Visit(syntax.CloseBracket);
             });
 
+        public override void VisitTypeDeclarationSyntax(TypeDeclarationSyntax syntax) =>
+            this.BuildStatement(syntax, () =>
+            {
+                this.VisitNodes(syntax.LeadingNodes);
+                this.Visit(syntax.Keyword);
+                this.documentStack.Push(Nil);
+                this.Visit(syntax.Name);
+                this.Visit(syntax.Assignment);
+                this.Visit(syntax.Value);
+            });
+
+        public override void VisitArrayTypeSyntax(ArrayTypeSyntax syntax) =>
+            this.BuildWithConcat(() => {
+                this.Visit(syntax.Item);
+                this.Visit(syntax.OpenBracket);
+                this.Visit(syntax.CloseBracket);
+            });
+
+        public override void VisitObjectTypeSyntax(ObjectTypeSyntax syntax) =>
+            this.BuildWithConcat(() => {
+                this.Visit(syntax.OpenBrace);
+                this.VisitCommaAndNewLineSeparated(syntax.Children, leadingAndTrailingSpace: true);
+                this.Visit(syntax.CloseBrace);
+            });
+
+        public override void VisitObjectTypePropertySyntax(ObjectTypePropertySyntax syntax) =>
+            this.BuildWithConcat(() =>
+            {
+                this.VisitNodes(syntax.LeadingNodes);
+                this.Visit(syntax.Key);
+                this.Visit(syntax.OptionalityMarker);
+                this.Visit(syntax.Colon);
+                this.documentStack.Push(Space);
+                this.Visit(syntax.Value);
+            });
+
```

Comment:
Personally I'd like to have space between union type members (`1 | 2 | 3`). Can we change the formatter to implement that? It can definately be done in a seperate PR though.

Reply:
The pretty printer should be doing this already (check out https://github.com/Azure/bicep/pull/8673/files#diff-21a3c4ce83e3e8c7ffadaf049301efb1d21374a0a8580f688344be771c64e10eR32 for an example of a formatted union).